### PR TITLE
Add a better Neon flow

### LINKED ISSUE #11: Leverage neonctl CLI to automatically grab/create a Neon project
It'd be really nice if, similarly to [neonctl's create-app command](https://neon.tech/docs/reference/cli-create-app), the `create-honc-app` leveraged `neonctl` to let users select one of their existing Neon projects _or_ create a new Neon project on the fly.

### PR DESCRIPTION
**Pull request by keturiosakys**
This PR introduces a feature for the`create-honc-app` CLI that enables authenticating with Neon and retrieving the project connection string automatically which should make it for a smoother setup flow.

resolves #11

### CODE REVIEW

File: cli/src/actions/database/neon.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        hint: `Last updated: ${database.updated_at}`,
+      })),
+    });
+
+    if (isCancel(database)) {
+      return database;
+    }
+
+    const role: string | symbol = await select({
+      message: "Select which role to use connecting to the database:",
```
Comment (brettimus) [inline cli/src/actions/database/neon.ts:97]: nit: "to connect to" or "when connecting"

File: cli/src/integrations/neon/auth.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      }
+
+      if (code) {
+        try {
+          const token = await client.validateAuthorizationCode(code, {
+            codeVerifier,
+          });
+          res.writeHead(200, { "Content-Type": "text/html" });
+          res.end(
+            "<code>Authentication with create-honc-app successful! You can close this window.</code>",
```
Comment (brettimus) [inline cli/src/integrations/neon/auth.ts:79]: goose? 🙈

**Review by brettimus — APPROVED** [review]

File: cli/src/integrations/neon/auth.ts (hunk 2 above)
Comment (keturiosakys) [reply cli/src/integrations/neon/auth.ts]: OBLIGATORY

Comment (keturiosakys) [reply cli/src/integrations/neon/auth.ts]: actually having tested it out, it won't work on plain text responses (punting it for the next update :))

**Pull request merged**